test/numbers.jl

```diff
@@ -1011,6 +1011,17 @@ f9085() = typemax(UInt64) != 2.0^64
 @test (1//typemax(Int)) / (1//typemax(Int)) == 1
 @test_throws OverflowError (1//2)^63
 
+for op in (+, -, *, /, ÷, %)
+    for (xfl, xra) in ((Inf, 1//0), (-Inf, -1//0)),
+        (yfl, yra) in ((Inf, 1//0), (-Inf, -1//0))
+        if isnan(op(xfl, yfl))
+            @test_throws Exception op(xra, yra)
```

Comment:
What kind of exception is most appropriate? Currently it throws either `ArgumentError` or `DivideError` depending on what goes wrong first, and I think that could be improved (if the overhead of a `try`-`catch` is acceptable, of course). I am thinking that `DomainError` makes most sense, since if the result would be `NaN` for floats that would mean the inputs are not in the domain of the function.